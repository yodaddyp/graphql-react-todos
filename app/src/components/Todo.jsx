import React from 'react';
import Checkbox from '@material-ui/core/Checkbox';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

export default function Todo({ completed, due, id, onCheck, text }) {
    const checked = !!completed;
    const labelId = `label-${id}`;
    const primary = text;
    const secondary = checked
      ? `Completed on ${completed}`
      : `Due on ${due}`;

    return (
        <ListItem role={undefined} dense button onClick={onCheck}>
            <ListItemIcon>
                <Checkbox
                    checked={checked}
                    edge="start"
                    inputProps={{ 'aria-labelledby': labelId }}
                    tabIndex={-1}
                />
            </ListItemIcon>
            <ListItemText
                id={labelId}
                primary={primary}
                secondary={secondary}
            />
        </ListItem>
    )
}