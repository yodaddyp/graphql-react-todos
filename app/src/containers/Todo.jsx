import React from 'react';
import { gql } from 'apollo-boost';
import { Mutation } from '@apollo/react-components';
import Todo from '../components/Todo';

const MARK_AS_DONE = gql`
mutation markAsDone($id: uuid!, $date: date) {
  update_todos(where: {id: {_eq: $id}}, _set: {completed: $date}) {
    returning {
      completed
      due
      id
      text
    }
  }
}
`;

export default props => (
    <Mutation mutation={MARK_AS_DONE}>
        {markAsDone => (
            <Todo
                {...props}
                onCheck={() => markAsDone({
                    variables: {
                        id: props.id,
                        date: props.completed ? null : new Date()
                    }
                })}
            />
        )}
    </Mutation>
)