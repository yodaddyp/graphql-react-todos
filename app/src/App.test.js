import React from 'react';
import { render } from '@testing-library/react';
import { defineFeature, loadFeature } from 'jest-cucumber';
import Todo from './components/Todo';

const feature = loadFeature('../acceptance/todos.feature', {
  tagFilter: '@unit'
});

defineFeature(feature, scenario => {
  scenario('I can see a todo', ({ given, when, then }) => {
    let actual;

    given('the following todo is defined:', table => {
      const props = table[0];
      actual = render(<Todo {...props} />)
    });

    then('it renders correctly', () => {
      expect(actual).toMatchSnapshot();
    });
  })
});

/*
test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
 */
