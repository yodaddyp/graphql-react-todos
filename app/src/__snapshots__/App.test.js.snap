// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos I can see a todo 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-23 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                aria-labelledby="label-1"
                class="PrivateSwitchBase-input-26"
                data-indeterminate="false"
                tabindex="-1"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
          id="label-1"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
          >
            my todo
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            Due on 2020-02-12
          </p>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-23 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              aria-labelledby="label-1"
              class="PrivateSwitchBase-input-26"
              data-indeterminate="false"
              tabindex="-1"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
        id="label-1"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
        >
          my todo
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Due on 2020-02-12
        </p>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
